cmd/coder/sync.go

```diff
@@ -29,6 +34,37 @@ func (cmd *syncCmd) RegisterFlags(fl *pflag.FlagSet) {
 	fl.BoolVarP(&cmd.init, "init", "i", false, "do initial transfer and exit")
 }
 
+// See https://lxadm.com/Rsync_exit_codes#List_of_standard_rsync_exit_codes.
+var IncompatRsync = errors.New("rsync: exit status 2")
+var StreamErrRsync = errors.New("rsync: exit status 12")
+
+// Returns local rsync protocol version as a string.
+func (s *syncCmd) version() string {
+	cmd := exec.Command("rsync", "--version")
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+
+	r := bufio.NewReader(stdout)
+	if err := cmd.Wait(); err != nil {
+		log.Fatal(err)
+	}
+
+	firstLine, err := r.ReadString('\n')
+	if err != nil {
+		return ""
+	}
+
+	versionString := strings.Split(firstLine, "protocol version ")
+
+	return versionString[1]
```

Comment:
The way this is laid out, `version` will always return an empty string. Reading from `r` will always error, because the pipe will be closed before we start reading from the command.

I would suggest replacing reading the output of the command with:
```
out, err := cmd.CombinedOutput()
if err != nil {
    log.Fatal(err)
}
```

And I would replace reading the first line with:
```
firstLine, err := bytes.NewBuffer(out).ReadString('\n')
if err != nil {
    log.Fatal(err)
}
```

Reply:
Will do.